### DISCUSSION TITLE
netbox ldap authentication issue

### DISCUSSION THREAD

**Discussion by lasag-2511** [netbox-community/netbox#15333] [Help Wanted!]
Hi all,

I wanted to use ldap authentication for the users in the ldap group "django-agent_group,"  and i followed the steps as below

import ldap
from django_auth_ldap.config import LDAPSearch, GroupOfNamesType

# Baseline configuration.
AUTH_LDAP_SERVER_URI = "ldap://ldap.example.com"

AUTH_LDAP_BIND_DN = "cn=django-agent,dc=example,dc=com"
AUTH_LDAP_BIND_PASSWORD = "phlebotinum"
AUTH_LDAP_REQUIRE_GROUP = "cn=django-agent_group,ou=Groups,o=com"

but the ldap authentication doesnot work at all. do you have suggestions to make ldap work.

AttributeError: 'tuple' object has no attribute -- this is the error i get when i login to homepage on netbox

**Comment by candlerb** [comment]
I think it's likely that you've put a comma at the end of a line somewhere in your configuration files. e.g.

```
FOO = "BAR",
```

will create a one-element tuple, as if you'd written

```
FOO = ("BAR", )
```

If you can generate a backtrace (e.g. by setting DEBUG=True) you may be able to ascertain which variable is affected. Otherwise, just look for trailing commas on values which are not inside a list `[...]` or a dict `{...}`
